x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/stats/Metrics.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.sql.stats;
+
+import org.elasticsearch.common.metrics.CounterMetric;
+import org.elasticsearch.xpack.core.watcher.common.stats.Counters;
+
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Map.Entry;
+
+/**
+ * Class encapsulating the metrics collected for ES SQL
+ */
+public class Metrics {
+    private enum OperationType {
+        FAILED, PAGING, TOTAL;
```

Comment:
This needs to be broken into `QUERY`/`PAGING` each one with `FAILED`/`SUCCESSFUL/TOTAL`